### PR TITLE
Clean up webconsole UI logic and remove dead code

### PR DESCRIPTION
**Pull request by markerikson**
This PR:

- Improves timeline perf by skipping dispatching the `setCanvas` action if the contents are already the same
- Migrates `webconsole/reducers/ui` to RTK+TS
- Removes a _bunch_ of dead code:
    - The only field in `reducers/ui` that was still being used as `timestampsVisible`, so I removed all other fields and associated action creators
    - The `prefsService` used in the thunks in `webconsole/actions/ui` was a stub, so I removed the references to that
    - `FilterBar` no longer has a "display mode"
    - Removed `webconsole/actions/ui` entirely as it had no remaining code
    - Removed numerous action type constants from `constants.js` that were no longer referenced anywhere

### CODE REVIEW

**Review by jasonLaster — COMMENTED** [review]
Looks good to me! Will let others give the official review.

File: src/devtools/client/webconsole/utils/prefs.js [hunk 1]
```diff
@@ -7,6 +7,8 @@
 import Services from "devtools/shared/services";
 import { PrefsHelper } from "devtools/client/shared/prefs";
 
+// TODO Can this file just be deleted entirely?
```
Comment (jaril) [inline src/devtools/client/webconsole/utils/prefs.js:10]: Eventually yes :)

File: src/devtools/client/webconsole/reducers/ui.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { createSlice, createNextState } from "@reduxjs/toolkit";
+
+interface WebconsoleUIState {
+  timestampsVisible: boolean;
+  // TODO Remove these dead fields and update `getVisibleMessages
```
Comment (jaril) [inline src/devtools/client/webconsole/reducers/ui.ts:8]: 👍

**Review by jaril — APPROVED** [review]
LGTM!

**Pull request merged**